lib/http/request/body.rb

```diff
@@ -33,8 +33,10 @@ def each
         if @body.is_a?(String)
           yield @body
         elsif @body.respond_to?(:read)
-          chunk = String.new # rubocop:disable Style/EmptyLiteral
-          yield chunk while @body.read(BUFFER_SIZE, chunk)
+          loop do
+            data = @body.read(BUFFER_SIZE) or break
```

Comment:
Personally I'd prefer something like:

```ruby
while (data = @body.read(BUFFER_SIZE)
  yield data
end
```

It better communicates intent, IMO, despite assigning in a branch.